phoenixRest/models/core/agenda_entry.py

```diff
@@ -31,21 +31,41 @@ class AgendaEntry(Base):
     event = relationship("Event")
     
     time = Column(DateTime, nullable=False)
+    newTime = Column(DateTime, nullable=False)
+
+    location = Column(Text, nullable=True)
+    newLocation = Column(Text, nullable=True)
 
     title = Column(Text, nullable=False)
     description = Column(Text, nullable=False)
+    modifyReason = Column(Text, nullable=True)
+    sticky = Column(Boolean, nullable=False)
 
-    def __init__(self, title: str, description: str, time: datetime, event: Event):
+    def __init__(self, title: str, description: str, event: Event, time: datetime, newTime: datetime, location: str, newLocation: str, modifyReason: str, sticky: bool):
         self.title = title
         self.description = description
         self.event = event
         self.time = time
+        self.newTime = newTime
+        self.location = location
+        self.newLocation = newLocation
+        self.modifyReason = modifyReason
+        self.sticky = sticky
 
     def __json__(self, request):
-        return {
+        data = {
             'uuid': str(self.uuid),
             'title': self.title,
             'description': self.description,
             'time': int(self.time.timestamp()),
-            'event_uuid': self.event_uuid
-        }
\ No newline at end of file
+            'location': self.location,
+            'newLocation': self.newLocation,
+            'modifyReason': self.modifyReason,
+            'sticky': self.sticky,
+            'event_uuid': self.event_uuid,
+        }
+
+        if self.newTime is not None:
```

Comment:
Avoid this logic, just use `int(self.newTime.timestamp()) if self.newTime is not None else None`.

An useful side-effect your code doesn't have is that this code will make the value of `newTime` `null` if there is no `newTime`, instead of the field not existing outright. There is a semantic difference between a field being `null` and a field being `undefined` which I think is useful.

Reply:
Changed logic for `time` and `deviating_time` to your suggestion.
Code tested in development environment, it works!

[Check this commit for the updated version!](https://github.com/phoenixlan/phoenixapi-v1/commit/484f3e969bfd1e8b7e8fab93a08b1cafbc23b739?diff=split)